src/tag-version.sh

```diff
@@ -40,9 +51,10 @@ getRevisionType() {
     outLog "(major, minor, build)";
     outLog "Message: $MESSAGE";
 
-    if [[ "$MESSAGE" == *"#major"* ]]; then
+    local INCREASE_MAJOR="$(isBreakingChange)"
+    if [[ ${INCREASE_MAJOR} == 0 ]]; then
         echo "major"
-    elif [[ "$MESSAGE" == *"#minor"* ]]; then
+    elif [[ "$MESSAGE" == "feat"* ]]; then
```

Comment:
Hi Do, as seen in this example, the feat can appear in one .

So you basically need to check for regex pattern .
```suggestion
    elif [[ "$MESSAGE" =~ '(^feat|[[:cntrl:]][[:blank:]]*feat)' ]]; then
```
:blank: will match spaces and tabs and cntrl will match new line or carriage return.  should work well with example below
```
commit 36ee749a23aaff069dc0ddcc00975a298cefd4e7 (tag: 0.146.14)
Merge: 724ea83f 3485fdb9
Author: Tam Nguyen <nguyenchitam1993@gmail.com>
Date:   Thu Dec 1 14:51:53 2022 +0700

    Merge pull request #360 from NU-LXP/feature/NU-74-remove-cloze-words
    
    feat(cloze-widget): can delete cloze words
```

Reply:
guys why are we making this so complex? there are industry standard solutions for this stuff, like `commitizen`, or , or what i recommend is the node package [`git-conventional-commits`](https://www.npmjs.com/package/git-conventional-commits) for which we already have a config file in the repo: https://github.com/NU-LXP/middleware-lms/blob/main/git-conventional-commits.json

Reply:
Hi @D3PSI , we analysed the code of git-conventional-cmmits here https://cleoag.atlassian.net/browse/DEVOPS-77. For doing a regex search on breaking commits or presence of `feat` verb we decided _not_ to add that complete npm library.  

I am also all for industry solutions, but with 2000monthly downloads for git-conventional-commits i would be really hesitate from calling that `standard` or even `industry`. 

After reviewing their code I dont see that it solves the versioning better. you can follow the links in DEVOPS-77 and see for yourself. Or do you think we miss anything?

Reply:
well with `commitizen` it looks bette. 700k downloads per month. but still dont see the point why we need to add that library for 2 regex matches?

Reply:
i wouldn't agree with this, especially the denial of service point. but i also wasn't part of this ticket and it's not my place to interfere here, so its totally fine for me if you guys want to maintain this stuff yourselves :+1:. i was just worried about the logic here: https://github.com/NU-LXP/common_workflows/pull/11#issuecomment-1337056167

Reply:
Hi @D3PSI , your comments are totally welcome and appreciated. 
I agree with you that we should favour existing solutions over developing our own ones. We are chronically suffering the "not invented by us" syndrom. Our whole organisation is perfect example. Instead of forking an existing LMS like moodle or edX we decided we know better and create our own 😮‍💨 . 
But please understand our needs at the dev ops team. We need to use tools that are stable: curl, awk, sed, jq, yq, ansible, rsync.. The fancy npm tools come with a lot of dependencies and a lot of security holes.|
I did not wrote with Do my own versioning tool from scrach but used git and bash tools that are more or less straight forward and got some code inipiration from https://github.com/julesabr/version-maestro/blob/master/src/version-maestro (as noted in the comment on top of our script)